torch/onnx/utils.py

```diff
@@ -137,23 +132,45 @@ def _export(model, args, f, export_params=True, verbose=False, training=False,
         raise RuntimeError("state_dict changed after running the tracer; "
                            "something weird is happening in your model!")
 
-    trace.set_graph(_optimize_graph(trace.graph(), aten))
+    return trace.graph(), torch_out
+
+
+# NOTE: the output `torch_out` will contain the output tensors resulting from
+# the trace of a Module. In the case that a torch.nn.ScriptModule is passed in,
+# this output will be None, since we are not doing any tracing but rather
+# directly extracting the graph.
+def _export(model, args, f, export_params=True, verbose=False, training=False,
+            input_names=None, output_names=None, aten=False, export_type=ExportTypes.PROTOBUF_FILE):
+    # Special case for common case of passing a single Variable
+    if isinstance(args, torch.autograd.Variable):
+        args = (args, )
+
+    if isinstance(model, torch.jit.ScriptModule):
+        torch_out = None
+        try:
+            graph = model.__getattr__('forward').graph()
```

Comment:
Why can't we just do `model.forward`?

Reply:
cc @zdevito 

model.forward seems to give me the actual Python forward method, but __getattr__('forward') gives me the ScriptMethod instance. Not sure why

Reply:
I think the safe thing to do here is to call `model._get_method('forward')` that will ensure you are always getting the Method instance. Otherwise you might get a python function.